src/effects/effectparameterslot.cpp

```diff
@@ -242,3 +253,54 @@ void EffectParameterSlot::slotValueChanged(double v) {
         m_pEffectParameter->setValue(v);
     }
 }
+
+QDomElement EffectParameterSlot::toXML(QDomDocument* doc) const {
+    QDomElement parameterElement;
+    if (m_pEffectParameter != nullptr) {
+        parameterElement = doc->createElement(EffectXml::Parameter);
+        XmlParse::addElement(*doc, parameterElement,
+                             EffectXml::ParameterId,
+                             m_pEffectParameter->id());
+        // TODO(rryan): Do smarter QVariant formatting?
+        XmlParse::addElement(*doc, parameterElement,
+                             EffectXml::ParameterValue,
+                             QString::number(m_pControlValue->getParameter()));
+        XmlParse::addElement(*doc, parameterElement,
+                             EffectXml::ParameterLinkType,
+                             EffectManifestParameter::LinkTypeToString(
+                                static_cast<EffectManifestParameter::LinkType>(
+                                    m_pControlLinkType->get())));
+        XmlParse::addElement(*doc, parameterElement,
+                             EffectXml::ParameterLinkInversion,
+                             QString::number(m_pControlLinkInverse->get()));
+    }
+
+    return parameterElement;
+}
+
+void EffectParameterSlot::loadValuesFromXml(const QDomElement& parameterElement) {
+    if (m_pEffectParameter == nullptr) {
+        return;
+    }
+    if (parameterElement.text().isEmpty()) {
+        m_pControlValue->reset();
+        m_pControlLinkType->set(
+            static_cast<double>(m_pEffectParameter->getDefaultLinkType()));
+        m_pControlLinkInverse->set(
+            static_cast<double>(m_pEffectParameter->getDefaultLinkInversion()));
+    } else {
+        // Need to use setParameterFrom(..., nullptr) here to
+        // trigger valueChanged() signal emission and execute slotValueChanged()
+        m_pControlValue->setParameterFrom(
+            XmlParse::selectNodeDouble(parameterElement,
+                                       EffectXml::ParameterValue),
+            nullptr);
+        m_pControlLinkType->set(static_cast<double>(
+            EffectManifestParameter::LinkTypeFromString(
+                XmlParse::selectNodeQString(parameterElement,
+                                            EffectXml::ParameterLinkType))));
+        m_pControlLinkInverse->set(
+            XmlParse::selectNodeDouble(parameterElement,
```

Comment:
This and the calls above can fail. I think it should go to default in this case, right?